// python3 -m grpc_tools.protoc -I./protos --python_out=. --pyi_out=. --grpc_python_out=. ./protos/transcriber_service.proto

syntax = "proto3";

service TranscriberService {
    rpc Transcribe(TranscribeRequest) returns (TranscribeResponse) {}
    rpc DetectLanguage(DetectLanguageRequest) returns (DetectLanguageResponse) {}
}

message TranscribeRequest {
    bytes audio_data = 1;
    string language = 2;
    string get_embedding = 3;
    float min_segment_duration = 4;
}

message TranscribeResponse {
    string segments = 1;
    repeated bytes embeddings = 2;
}

message DetectLanguageRequest {
    bytes audio_data = 1;
}

message DetectLanguageResponse {
    string detected_languages = 1;
}
